src/main/java/com/onarandombox/multiverseinventories/util/CommentedYamlConfiguration.java

```diff
@@ -160,116 +155,87 @@ public boolean save() {
                                 // If there is a final period, replace everything after it with nothing
                                 currentPath.replace(currentPath.lastIndexOf("."), currentPath.length(), "").append(".");
                             }
-                            //currentPath = currentPath.replace(currentPath.substring(currentPath.lastIndexOf(".")), "");
-                            currentPath.append(line.substring(whiteSpace, index));
+
+                            currentPath.append(line, whiteSpace, index);
                         }
                     }
                 } else {
                     node = false;
                 }
+
                 StringBuilder newLine = new StringBuilder(line);
```

Comment:
Will it be a better idea to have an empty `StringBuilder` here so we use `newLine.append(...)` instead of `newLine.insert(0, ...)` below? iirc append has better time complexity than insert assuming StringBuilder backend logic is similar to arraylist

Reply:
Just searched it up, you're right. Insert's time complexity is worse than append's.

I'm trying some stuff now to reduce the usage of insert.

Reply:
Yea you should be able to just add the comment lines before the config line, should remove the need for insert for that part.